File: 2026/06-June/02-Tuesday/solution.cpp
class Solution {
public:
    int calc(vector<int>& firstStart, vector<int>& firstDur,
             vector<int>& secondStart, vector<int>& secondDur) {

        int ans = INT_MAX;

        for (int i = 0; i < firstStart.size(); i++) {
            int finishFirst = firstStart[i] + firstDur[i];

            for (int j = 0; j < secondStart.size(); j++) {
                int startSecond = max(finishFirst, secondStart[j]);
                int finishSecond = startSecond + secondDur[j];

                ans = min(ans, finishSecond);
            }
        }

        return ans;
    }

    int earliestFinishTime(vector<int>& landStartTime,
                           vector<int>& landDuration,
                           vector<int>& waterStartTime,
                           vector<int>& waterDuration) {

        return min(
            calc(landStartTime, landDuration,
                 waterStartTime, waterDuration),

            calc(waterStartTime, waterDuration,
                 landStartTime, landDuration)
        );
    }
};

File: 2026/06-June/02-Tuesday/solution.py
class Solution:
    def calc(self, first_start, first_dur, second_start, second_dur):
        ans = float('inf')

        for i in range(len(first_start)):
            finish_first = first_start[i] + first_dur[i]

            for j in range(len(second_start)):
                start_second = max(finish_first, second_start[j])
                finish_second = start_second + second_dur[j]

                ans = min(ans, finish_second)

        return ans

    def earliestFinishTime(self, landStartTime, landDuration,
                           waterStartTime, waterDuration):

        return min(
            self.calc(
                landStartTime,
                landDuration,
                waterStartTime,
                waterDuration
            ),
            self.calc(
                waterStartTime,
                waterDuration,
                landStartTime,
                landDuration
            )
        )
